tfe/provider.go

```diff
@@ -221,6 +251,111 @@ func credentialsSource(config *Config) auth.CredentialsSource {
 	return creds
 }
 
+// checkConstraints checks service version constrains against our own
+// version and returns rich and informational diagnostics in case any
+// incompatibilities are detected.
+func checkConstraints(c *disco.Constraints) error {
```

Comment:
It would be nice to have a test for this, but it's not a blocker.

Reply:
Yeah agree, so added a test for it.